src/org/labkey/targetedms/query/PeptideManager.java

```diff
@@ -115,10 +120,68 @@ public static List<PeptideCharacteristic> getPeptideCharacteristic(long peptideG
         sql.append(" ON gm.id = pep.Id");
         sql.append(" INNER JOIN ").append(TargetedMSManager.getTableInfoSampleFile(), "sf");
         sql.append(" ON sf.Id = pci.SampleFileId");
+        if (!Objects.equals(replicateId, Long.valueOf(0)))
+        {
+            sql.append(" INNER JOIN ").append(TargetedMSManager.getTableInfoReplicate(), "rep");
+            sql.append(" ON rep.Id = sf.ReplicateId");
+        }
+        sql.append(" WHERE gm.PeptideGroupId=? ");
+        if (!Objects.equals(replicateId, Long.valueOf(0)))
+        {
+            sql.append(" AND rep.Id=? ");
+        }
+        sql.append(" GROUP BY pep.Sequence,pci.SampleFileId, pep.StartIndex, pep.EndIndex ) X ");
+        sql.append(" GROUP BY X.Sequence, X.StartIndex, X.EndIndex ");
+        sql.add(peptideGroupId);
```

Comment:
I find it easier and more maintainable to add the value closer to the line that includes the `?` in the SQL. As in, shift this up to right after calling `sql.append(" WHERE gm.PeptideGroupId=? ");`